props/blaster.h

```diff
@@ -0,0 +1,192 @@
+#ifndef PROPS_BLASTER_H
+#define PROPS_BLASTER_H
+
+#include "prop_base.h"
+
+#define PROP_TYPE Blaster
+
+class Blaster : public PropBase {
+public:
+  Blaster() : PropBase() {}
+  const char* name() override { return "Blaster"; }
+
+#ifdef DELAYED_OFF
+  bool powered_ = false;
+  void SetPower(bool on) { powered_ = on; }
+#else
+  constexpr bool powered_ = true;
+  void SetPower(bool on) {}
+#endif
+
+  // Mode states to handle kill vs stun effects
+  enum BlasterMode {
+    MODE_STUN,
+    MODE_KILL,
+    MODE_AUTO
+  };
+
+  BlasterMode blaster_mode = MODE_STUN;
+
+  void SetBlasterMode(BlasterMode to_mode) {

```

Comment:
make it virtual to make it easy for people to inherit and extend, same for nextblastermode and fire

Reply:
done.